flink-connector-aws-base/src/main/java/org/apache/flink/connector/aws/config/AWSConfigConstants.java

```diff
@@ -144,6 +145,194 @@ public enum CredentialProvider {
+     *
+     * <p>In default, this is disabled.
+     *
+     * <p>For more details, see
+     * https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.Builder.html
+     */
+    public static final String HTTP_CLIENT_APACHE_EXPECT_CONTINUE_ENABLED =
+            "http-client.apache.expect-continue-enabled";
+
+    /**
+     * Used to configure the max connections number for {@link
+     * software.amazon.awssdk.http.apache.ApacheHttpClient.Builder}. This flag only works when
+     * {@link #HTTP_CLIENT_TYPE} is set to HTTP_CLIENT_TYPE_APACHE
+     *
+     * <p>For more details, see
+     * https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.Builder.html
+     */
+    public static final String HTTP_CLIENT_APACHE_MAX_CONNECTIONS =
+            "http-client.apache.max-connections";
+
+    /**
+     * Used to configure the socket timeout in milliseconds for {@link
+     * software.amazon.awssdk.http.apache.ApacheHttpClient.Builder}. This flag only works when
+     * {@link #HTTP_CLIENT_TYPE} is set to HTTP_CLIENT_TYPE_APACHE
+     *
+     * <p>For more details, see
+     * https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.Builder.html
+     */
+    public static final String HTTP_CLIENT_APACHE_SOCKET_TIMEOUT_MS =
+            "http-client.apache.socket-timeout-ms";
+
+    /**
+     * Used to configure whether to enable the tcp keep alive setting for {@link
+     * software.amazon.awssdk.http.apache.ApacheHttpClient.Builder}. This flag only works when
+     * {@link #HTTP_CLIENT_TYPE} is set to HTTP_CLIENT_TYPE_APACHE.
+     *
+     * <p>In default, this is disabled.
+     *
+     * <p>For more details, see
+     * https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.Builder.html
+     */
+    public static final String HTTP_CLIENT_APACHE_TCP_KEEP_ALIVE_ENABLED =
+            "http-client.apache.tcp-keep-alive-enabled";
+
+    /**
+     * Used to configure whether to use idle connection reaper for {@link
+     * software.amazon.awssdk.http.apache.ApacheHttpClient.Builder}. This flag only works when
+     * {@link #HTTP_CLIENT_TYPE} is set to HTTP_CLIENT_TYPE_APACHE.
+     *
+     * <p>In default, this is enabled.
+     *
+     * <p>For more details, see
+     * https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.Builder.html
+     */
+    public static final String HTTP_CLIENT_APACHE_USE_IDLE_CONNECTION_REAPER_ENABLED =
+            "http-client.apache.use-idle-connection-reaper-enabled";
+
+    /**
+     * Configure an alternative endpoint of the Glue service for GlueCatalog to access.
+     *
+     * <p>This could be used to use GlueCatalog with any glue-compatible metastore service that has
+     * a different endpoint
+     */
+    public static final String GLUE_CATALOG_ENDPOINT = "glue.endpoint";
```

Comment:
We mainly use this package for common AWS utils and constants, this seems Glue specific IMO